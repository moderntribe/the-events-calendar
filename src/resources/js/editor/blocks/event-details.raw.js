( function( $ ) {
	const { Component } = wp.element;
	const { Placeholder, Spinner } = wp.components;
	const { __ } = wp.i18n;
	const { registerBlockType } = wp.blocks;


	var requestAjax = function ( action, data ) {
		var request = {
			dataType: 'json',
			method: 'POST',
			url: window.ajaxurl + '?action=' + encodeURIComponent( action ),
			async: false,
			data: data
		};

		return JSON.parse( $.ajax( request ).responseText );
	};


	class EventDetails extends Component {
		componentDidMount() {
			this.request();
		}

		componentDidUpdate( previousProps ) {
			const { modelName, query } = this.props;
			if (
				modelName !== previousProps.modelName ||
				! _.isEqual( query, previousProps.query )
			) {
				this.request();
			}
		}

		request() {

		}

		render() {
			const { focus } = this.props;

			var response = requestAjax( 'tribe_editor_block_event-details', { test: 'dev' } );
			var html = '';

			if ( response.success ) {
				html = response.data;
			}

			return (
				<div className="tribe-editor-block-wrap">
					{ HTMLReactParser( html ) }
				</div>
			);//

		}
	}//

	var blockName = 'tribe/event-details';
	var blockConfiguration = {
		title: __( 'Event Details', 'the-events-calendar' ),
		description: __( 'Configuration for the Event', 'the-events-calendar' ),
		icon: 'calendar',
		category: 'layout',
		keywords: [ 'event', 'the-events-calendar', 'tribe' ],

		useOnce: true,

		attributes: {
		},

		// The "edit" property must be a valid function.
		edit: function( props ) {
			return <EventDetails />;
		},

		// The "save" property must be specified and must be a valid function.
		save: function( props ) {
			return null;
		},
	};

	// Actually Register the block on Editor
	registerBlockType( blockName, blockConfiguration );
} )( jQuery );