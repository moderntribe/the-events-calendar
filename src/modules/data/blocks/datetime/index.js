/**
 * Internal dependencies
 */
import reducer, { setInitialState, defaultStateToMetaMap } from './reducer';
import * as types from './types';
import * as actions from './actions';
import * as selectors from './selectors';
import sagas from './sagas';

export default reducer;
export { types, actions, selectors, sagas, setInitialState, defaultStateToMetaMap };
