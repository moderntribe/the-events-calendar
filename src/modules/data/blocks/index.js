/**
 * Internal dependencies
 */
import * as datetime from './datetime';
import subscribe from './subscribers';

export { datetime, subscribe };
export { default, setInitialState } from './reducer';
