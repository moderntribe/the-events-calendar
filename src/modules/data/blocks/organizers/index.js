/**
 * Internal dependencies
 */
import reducer, { setInitialState } from './reducer';
import * as types from './types';
import * as actions from './actions';
import * as selectors from './selectors';
import subscribe from './subscribers';

export default reducer;
export { types, actions, selectors, setInitialState, subscribe };
