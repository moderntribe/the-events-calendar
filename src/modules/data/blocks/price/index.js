/**
 * Internal dependencies
 */
import reducer, { setInitialState, defaultStateToMetaMap } from './reducer';

import * as selectors from './selectors';
import * as actions from './actions';
import * as types from './types';
import * as utils from './utils';

export default reducer;
export { selectors, actions, types, utils, setInitialState, defaultStateToMetaMap };
