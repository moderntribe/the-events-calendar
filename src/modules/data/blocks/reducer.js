/**
 * External dependencies
 */
import { combineReducers } from 'redux';

/**
 * Internal dependencies
 */
import datetime, { setInitialState as datetimeSetInitialState } from './datetime';
import organizers, { setInitialState as organizersSetInitialState } from './organizers';
import price, { setInitialState as priceSetInitialState } from './price';
import website, { setInitialState as websiteSetInitialState } from './website';
import venue, { setInitialState as venueSetInitialState } from './venue';

export const setInitialState = ( data ) => {
	datetimeSetInitialState( data );
	organizersSetInitialState( data );
	priceSetInitialState( data );
	venueSetInitialState( data );
	websiteSetInitialState( data );
};

export default combineReducers( {
	datetime,
	venue,
	organizers,
	price,
	website,
} );
