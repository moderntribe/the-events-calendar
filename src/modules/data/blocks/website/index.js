/**
 * Internal dependencies
 */
import reducer, { setInitialState, defaultStateToMetaMap } from './reducer';

import * as selectors from './selectors';
import * as actions from './actions';
import * as types from './types';

export default reducer;
export { selectors, actions, types, setInitialState, defaultStateToMetaMap };
