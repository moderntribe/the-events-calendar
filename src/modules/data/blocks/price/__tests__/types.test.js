/**
 * Internal dependencies
 */
import { types } from '@moderntribe/events/data/blocks/price';

describe( '[STORE] - Price types', () => {
	it( 'Should return the types values', () => {
		expect( types.SET_PRICE_POSITION ).toMatchSnapshot();
		expect( types.SET_PRICE_SYMBOL ).toMatchSnapshot();
		expect( types.SET_PRICE_COST ).toMatchSnapshot();
	} );
} );
