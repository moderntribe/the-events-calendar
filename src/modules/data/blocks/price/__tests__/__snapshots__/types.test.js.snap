// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[STORE] - Price types Should return the types values 1`] = `"@@MT/EVENTS/SET_PRICE_POSITION"`;

exports[`[STORE] - Price types Should return the types values 2`] = `"@@MT/EVENTS/SET_PRICE_SYMBOL"`;

exports[`[STORE] - Price types Should return the types values 3`] = `"@@MT/EVENTS/SET_PRICE_COST"`;
